eclipse-scout-core/src/calendar/Calendar.ts

```diff
@@ -481,12 +493,85 @@ export class Calendar extends Widget implements CalendarModel {
 
   protected override _renderProperties() {
     super._renderProperties();
+    this._renderCalendars();
     this._renderComponents();
     this._renderSelectedComponent();
     this._renderLoadInProgress();
     this._renderDisplayMode();
   }
 
+  setCalendars(calendars: CalendarDescriptor[]) {
+    this.setProperty('calendars', calendars);
+  }
+
+  protected _setCalendars(calendars: CalendarDescriptor[]) {
+    this._setProperty('calendars', calendars);
+    this._updateCalendarsPanelDisplayable();
+    this._updateCalendarsPanel();
+    this._validateSelectedCalendar();
+  }
+
+  protected _renderCalendars() {
+    let $dayName = this.$topGrid.find('.calendar-week-header > .calendar-day-name');
+    let $fullDay = this.$topGrid.find('.calendar-week-allday-container > .calendar-day');
+    let $day = this.$grid.find('.calendar-week > .calendar-day');
+
+    // Remove old calendar columns
+    $dayName.find('.calendar-column').remove();
+    $fullDay.find('.calendar-column').remove();
+    $day.find('.calendar-column').remove();
+
+    // Add default calendar columns
+    $dayName.appendDiv('calendar-column')
+      .data('calendarId', this.defaultCalendar.calendarId);
+    $fullDay.appendDiv('calendar-column')
+      .data('calendarId', this.defaultCalendar.calendarId);
+    $day.appendDiv('calendar-column')
+      .data('calendarId', this.defaultCalendar.calendarId);
+
+    // Add new calendar columns
+    this._getCalendarsWithoutGroups().forEach(calendar => {
+      $dayName.appendDiv('calendar-column')
+        .data('calendarId', calendar.calendarId)
+        .attr('data-calendar-name', calendar.name);
+      $fullDay.appendDiv('calendar-column')
+        .data('calendarId', calendar.calendarId);
+      $day.appendDiv('calendar-column')
+        .data('calendarId', calendar.calendarId);
+    });
+
+    // click event on all day and children elements
+    let mousedownCallbackWithTime = this._onDayColumnMouseDown.bind(this, true);
+    this.$grid.find('.calendar-column').on('mousedown', mousedownCallbackWithTime);
+    let mousedownCallback = this._onDayColumnMouseDown.bind(this, false);
+    this.$topGrid.find('.calendar-column').on('mousedown', mousedownCallback);
+
+    this._renderComponents();
```

Comment:
It is necessary, because when the setCalendarDescriptors is called, the calendars are re-rendered. So do have the components be re-rendered. It is not nescessary in the initial render process. So I wrapped the render statement in an if-clause, to check if `this.rendering = true`.